src/engine/GroupBy.h

```diff
@@ -209,12 +216,14 @@ class GroupBy : public Operation {
   // Stores information required for substitution of grouped by variable in
   // an expression tree.
   struct GroupedByVariableSubstitutions {
-    std::vector<ParentAndChildIndex> occurrences;
-    bool topLevel;
+    std::vector<ParentAndChildIndex> occurrences_;
+    // Determines whether the grouped by variable appears at the top of an
+    // alias, e.g. `SELECT (?a as ?x) WHERE {...} GROUP BY ?a`.
+    bool topLevel_;
```

Comment:
Ah, 
But then it is *either* at thte toplevel or there are occurences, so something like
```
struct OccurenceAsRoot{}
std::variant<std::vector<ParentAndChildIndex>, OccurenceAsRoot> occurences_;
```
(The empty struct is just a tag to figure out the case. Your current ccode looks like `topLevel == true` can be combined with some occurences.